pkg/retry/retry.go

```diff
@@ -61,7 +62,12 @@ func (s *Server) handleRetry() http.Handler {
 			return
 		}
 
-		for cursor != "" {
+		logger.Infow("retrieved last checkpoint", "checkpoint", cursor)
+
+		// for each run track the total number of events retrieve and failure events
+		totalEventCount, failedEventCount := 0, 0
+
+		for ok := true; ok; ok = (cursor != "") {
```

Comment:
@sethvargo i brought this back because we need the loop to run at least once. The first time this service is invoked, the checkpoint will be empty. I'm attempting to replicate a do while loop in go. If you agree, is this the correct way?

Reply:
Yea this is common. The other option is to `break` under a condition, but this is fine.